common/src/main/java/noobanidus/mods/lootr/common/block/entity/BlockEntityTicker.java

```diff
@@ -20,195 +20,130 @@
 import noobanidus.mods.lootr.common.api.LootrTags;
 import noobanidus.mods.lootr.common.api.PlatformAPI;
 import noobanidus.mods.lootr.common.api.data.blockentity.ILootrBlockEntity;
+import org.jetbrains.annotations.Nullable;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
 import java.util.Set;
 
 public class BlockEntityTicker {
-  private final static Object listLock = new Object();
-  private final static Object worldLock = new Object();
   private final static Set<Entry> blockEntityEntries = new ObjectOpenHashSet<>();
   private final static Set<Entry> pendingEntries = new ObjectOpenHashSet<>();
-  private static boolean tickingList = false;
 
-  public static void addEntry(RandomizableContainerBlockEntity incoming, Level level, BlockPos position) {
+  public static void addChunkEntities(Collection<BlockEntity> entities, Level level, ChunkPos chunkPos) {
```

Comment:
I try to batch a whole chunk loading into one call to make things more efficient. Some checks are per-chunk and can be done for all entities at once here.